scripts/visualize.py

```diff
@@ -186,6 +196,19 @@ def animate_func(self, i):
             pos2 = pos0 + np.array([np.cos(theta0), np.sin(theta0)])*self.size[0]/2*0.8
             self.robot_patches[k][2].center = pos2
 
+      if(self.visualize_constraints):
+        if len(self.constraints["constraints"]) > k: # not all robots have constraints
+          if any(i == time * 10 for time in self.constraints["constraints"][k]["time"]):
+            index = (self.constraints["constraints"][k]["time"]).index(i / 10)
+            pos = self.constraints["constraints"][k]["states"][index]
+            xy = np.asarray(pos[:2])
+            patch_id = k + self.robot_numbers
+            self.robot_patches[patch_id][0].set(visible=True)
+            self.robot_patches[patch_id][0].center = xy
```

Comment:
Don't we sometimes have multiple constraints per timestep? At least for cases with many high-level expansions that's certainly possible.

Reply:
yes, with the latest/pushed changes, we don't have to care about it, thanks, nice catch!